fs/vfs/fs_close.c

```diff
@@ -84,6 +122,14 @@ int file_close_without_clear(FAR struct file *filep)
 
       if (ret >= 0)
         {
+#ifdef CONFIG_FS_NOTIFY
+          if (path != NULL)
+            {
+              notify_close(path, filep->f_oflags);
```

Comment:
fail in close callback is very rare, so let's move the notify_close before line 110 to simply the code logic?

Reply:
I don't know. It is rare, but still may occur. Moreover, we would effectively be reporting the close (and posting the notification) before it occurs, which might be dangerous as the notificated thread (lets say it preempts the thread calling `close()`) may expect the file is already closed.

It adds some code to `fs_close.c`, but removes the same code from `inotify.c`, so the performance is more or less the same.